### PR TITLE
[ESN-1660]WAF Harvest Fix

### PR DESCRIPTION
**Pull request by javier-tello23**
## Description

This PR updates the `get_package_dict` function to support WAF harvests and adds unit tests for this functionality.

## Test Procedure

**UI Testing**
1. Checkout the branch for this PR and run `python setup.py develop`
2. In your ini:
 - Disable the plugin `xloader` (This is done to save storage since the harvest url has 185 datasets)
 - Enable the following plugins
```
waf_harvester
spatial_metadata
spatial_query
cnra_schema
scheming_datasets
composite
repeating
```
3. Checkout the `cnra` branch in `ckanext-og_theme` and run `python setup.py develop`.
4. Restart supervisor with the following command: `sudo service supervisor restart`
5. Start CKAN and navigate to the harvest page `http://127.0.0.1:5000/harvest`.
6. Create a new job using the following details
 - URL: https://gis.water.ca.gov/app/test/wafDatagovDOI-orig/ or https://gis.water.ca.gov/app/test/wafDatagovDOI-translated/
 - Source type: Web Accessible Folder (WAF)
 - Configuration:
```
{
"publisher": {"publisher_field": "publisher"},
"default_tags":  ["data.gov", "doi-gov"], 
"map_fields": [{"default": "English", "source": "language", "target": "language"}, {"default": "", "source": "spatial", "target": "spatial_coverage"}], 
"default_groups": ["water"], 
"default_values": [{"public_access_level": "Public"}], 
"contact_point": {"name_field": "contact_name", "default_name": "None", "email_field": "contact_email", "default_email": "None"}
}
```
7. Run the harvest job and ensure datasets are added. 

**Unit Testing**
In the `ckanext-cnra_schema` directory, run the following command:  `nosetests --ckan --with-pylons=test.ini ckanext/cnra_schema/tests`

### CODE REVIEW

**Comment by jguo144** [comment]
The url https://gis.water.ca.gov/app/test/wafDatagovDOI-orig/ that was provided as a test does not work.
The example configuration that was provided as a test also does not work. It is missing a mapping for the field `progress`, which appears both in the extras field and package schema.

I was able to harvest the urls https://gis.water.ca.gov/app/test/wafDatagovDOI-translated/ and http://data.nanoos.org/metadata/coastwatcherded/osuclm/ with the following config
```
{
  "publisher": {"publisher_field": "publisher"},
  "default_tags":  ["data.gov", "doi-gov"], 
  "map_fields": [
    {"source": "language", "target": "language", "default": "English"},
    {"source": "spatial", "target": "spatial_coverage", "default": ""},
    {"source": "progress", "target": "progress", "default": ""}
  ],
  "default_groups": ["water"], 
  "default_values": [{"public_access_level": "Public"}], 
  "contact_point": {
    "name_field": "contact_name", "default_name": "None",
    "email_field": "contact_email", "default_email": "None"
  }
}
```

**Comment by javier-tello23** [comment]
It looks like I forgot to add `{"source": "progress", "target": "progress", "default": ""}` to map_fields for the config file and it everything is running fine. I have also tested using the following url http://52.55.221.16/WAF/caricoos/iso/ and the harvest job ran fine.

File: ckanext/cnra_schema/plugins.py [hunk 1]
```diff
@@ -26,6 +29,19 @@ def update_config(self, config):
...
     def get_package_dict(self, context, data_dict):
+        harvest_object = data_dict['harvest_object']
+        harvest_source_type = harvest_object.source.type
+
+        if harvest_source_type == 'fgdc':
+            modified_package_dict = self.get_fgdc_package_dict(data_dict)
+
+        elif harvest_source_type == 'waf':
+            modified_package_dict = self.get_waf_package_dict(data_dict)
+
```
Comment (jguo144) [inline ckanext/cnra_schema/plugins.py:40]: If `harvest_source_type` is not either fgdc or waf then `modified_package_dict` will be undefined.
You should return `data_dict['package_dict']` if `harvest_source_type` in not fgdc or waf.

File: ckanext/cnra_schema/plugins.py [hunk 2]
```diff
@@ -26,6 +29,19 @@ def update_config(self, config):
...
+        harvest_object = data_dict['harvest_object']
+        harvest_source_type = harvest_object.source.type
+
+        if harvest_source_type == 'fgdc':
+            modified_package_dict = self.get_fgdc_package_dict(data_dict)
+
+        elif harvest_source_type == 'waf':
+            modified_package_dict = self.get_waf_package_dict(data_dict)
+        
+        return modified_package_dict
```
Comment (jguo144) [inline ckanext/cnra_schema/plugins.py:41]: If `harvest_source_type` is not either fgdc or waf then `modified_package_dict` will be undefined.
You should return `data_dict['package_dict']` if `harvest_source_type` in not fgdc or waf.

File: ckanext/cnra_schema/plugins.py [hunk 3]
```diff
@@ -26,6 +29,21 @@ def update_config(self, config):
 """
 
     def get_package_dict(self, context, data_dict):
+        harvest_object = data_dict['harvest_object']
+        harvest_source_type = harvest_object.source.type
+
+        if harvest_source_type == 'fgdc':
```
Comment (jguo144) [inline ckanext/cnra_schema/plugins.py:35]: I've tested the CSW FGDC harvester and the source type should be set to `csw_fgdc`.
https://github.com/OpenGov-OpenData/ckanext-spatial/blob/master/ckanext/spatial/harvesters/csw_fgdc.py#L94

Comment (javier-tello23) [reply ckanext/cnra_schema/plugins.py]: You're right, I just finished testing and confirmed that the source type needs to be `csw_fgdc`. I have pushed this change

**Review by jguo144 — APPROVED** [review]
LGTM

**Pull request merged**